core/lib/l1_contract_interface/src/i_executor/commit/kzg.rs

```diff
@@ -0,0 +1,410 @@
+    kzg::{compute_commitment, compute_proof, compute_proof_poly, KzgSettings},
+    zkevm_circuits::{
+        boojum::pairing::{
+            bls12_381::{Fr, FrRepr, G1Affine},
+            CurveAffine,
+        },
+        eip_4844::{
+            bitreverse, fft,
+            input::{BLOB_CHUNK_SIZE, ELEMENTS_PER_4844_BLOCK},
+            zksync_pubdata_into_ethereum_4844_data, zksync_pubdata_into_monomial_form_poly,
+        },
+    },
+};
+
+const ZK_SYNC_BYTES_PER_BLOB: usize = BLOB_CHUNK_SIZE * ELEMENTS_PER_4844_BLOCK;
+const EIP_4844_BYTES_PER_BLOB: usize = 32 * ELEMENTS_PER_4844_BLOCK;
+
+/// Packed pubdata commitments.
+/// Format: opening point (16 bytes) || claimed value (32 bytes) || commitment (48 bytes)
+///         || opening proof (48 bytes)) = 144 bytes
+const BYTES_PER_PUBDATA_COMMITMENT: usize = 144;
+
+const VERSIONED_HASH_VERSION_KZG: u8 = 0x01;
+
+/// All the info needed for both the network transaction and by our L1 contracts. As part of the network transaction we
+/// need to encode the sidecar which contains the: blob, `kzg` commitment, and the blob proof. The transaction payload
+/// will utilize the versioned hash. The info needed for `commitBatches` is the `kzg` commitment, opening point,
+/// opening value, and opening proof.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct KzgInfo {
+    /// 4844 Compatible blob containing pubdata
+    pub blob: [u8; EIP_4844_BYTES_PER_BLOB],
+    /// KZG commitment to the blob
+    pub kzg_commitment: [u8; 48],
+    /// Point used by the point evaluation precompile
+    pub opening_point: [u8; 32],
+    /// Value retrieved by evaluation the `kzg` commitment at the `opening_point`
+    pub opening_value: [u8; 32],
+    /// Proof that opening the `kzg` commitment at the opening point yields the opening value
+    pub opening_proof: [u8; 48],
+    /// Hash of the `kzg` commitment where the first byte has been substituted for `VERSIONED_HASH_VERSION_KZG`
+    pub versioned_hash: [u8; 32],
+    /// Proof that the blob and `kzg` commitment represent the same data.
+    pub blob_proof: [u8; 48],
+}
+
+/// Given a KZG commitment, calculate the versioned hash.
+fn commitment_to_versioned_hash(kzg_commitment: G1Affine) -> [u8; 32] {
+    let mut versioned_hash = [0u8; 32];
+
+    let mut versioned_hash_bytes = Sha256::digest(kzg_commitment.into_compressed());
+    versioned_hash_bytes[0] = VERSIONED_HASH_VERSION_KZG;
+
+    versioned_hash.copy_from_slice(&versioned_hash_bytes);
+    versioned_hash
+}
+
+/// Calculate the opening point for a given `linear_hash` and `versioned_hash`. We calculate
+/// this point by hashing together the linear hash and versioned hash and only taking the last 16 bytes
+fn compute_opening_point(linear_hash: [u8; 32], versioned_hash: [u8; 32]) -> u128 {
+    let evaluation_point = &Keccak256::digest(
+        linear_hash
+            .iter()
+            .chain(&versioned_hash)
+            .copied()
+            .collect::<Vec<u8>>(),
```

Comment:
Can this be [`[linear_hash, versioned_hash].concat()`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=525f9c4ee1548ea8967bc913caac9b64)?